src/utils/conversion.ts

```diff
@@ -2288,7 +2288,8 @@ export function middlewareEventDetailsToEventIdentifier(
  * @hidden
  */
 export function meshClaimToClaim(claim: PolymeshPrimitivesIdentityClaimClaim): Claim {
-  if (claim.isJurisdiction) {
+  const type = claim.type;
+  if (type === 'Jurisdiction') {
```

Comment:
Looking at the type lets the UnreachableCaseError work. The main advantage is if a new claim gets added we will get a compile error to update this function, instead of mistaking it for a Blocked claim.